### PR TITLE
5105/web3 migration guide

### LINKED ISSUE #5105: Migration Guide/Breaking Changes - Web3
This information relates to the API documentation.

### PR DESCRIPTION
**Pull request by nikoulai**
## Description

#5105

### CODE REVIEW

File: docs/docs/guides/web3_migration_guide.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+-   [extend](https://web3js.readthedocs.io/en/v1.7.3/web3.html#extend) Extending web3 modules functionality is not implemented
+-   [web3.bzz](https://web3js.readthedocs.io/en/v1.7.3/web3-bzz.html) Package for interacting with Swarm is not implemented
+-   [web3.shh](https://web3js.readthedocs.io/en/v1.7.3/web3-shh.html) Package for interacting with Whisper is not implemented
+-   [web3.net](https://web3js.readthedocs.io/en/v1.7.3/web3-eth.html#net) namespace is not exported
+
+### Defaults and Configs
+
+-   `givenProvider` default value is `undefined` instead of `null`
+-   `currentProvider` will never return `null`, provider required upon instantiation as opposed to being optional in 1.x
```
Comment (nazarhussain) [inline docs/docs/guides/web3_migration_guide.md:15]: With the recent PR https://github.com/ChainSafe/web3.js/pull/5139 merged, the provider is optional now.

Comment (nazarhussain) [inline docs/docs/guides/web3_migration_guide.md]: With the recent PR https://github.com/ChainSafe/web3.js/pull/5139 merged, the provider is optional now.

File: docs/docs/guides/web3_migration_guide.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+### Web3 Instance
+
+provider parameter is required in 4.x
```
Comment (nazarhussain) [inline docs/docs/guides/web3_migration_guide.md]: With the recent PR https://github.com/ChainSafe/web3.js/pull/5139 merged, the provider is optional now.

File: docs/docs/guides/web3_migration_guide.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
+# Web3 Migration Guide
+
+## Breaking Changes
+
+### Not Implemented or Exported
+
+-   [extend](https://web3js.readthedocs.io/en/v1.7.3/web3.html#extend) Extending web3 modules functionality is not implemented
+-   [web3.bzz](https://web3js.readthedocs.io/en/v1.7.3/web3-bzz.html) Package for interacting with Swarm is not implemented
+-   [web3.shh](https://web3js.readthedocs.io/en/v1.7.3/web3-shh.html) Package for interacting with Whisper is not implemented
+-   [web3.net](https://web3js.readthedocs.io/en/v1.7.3/web3-eth.html#net) namespace is not exported
```
Comment (nazarhussain) [inline docs/docs/guides/web3_migration_guide.md]: It's exported.

File: docs/docs/guides/web3_migration_guide.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+in 1.x nothing was returned, in 4.x returns a deferred promise which can be used to run specific code after completion of each respective request
+
+```ts
+// in 1.x
+batch.add(
+	web3.eth.getBalance.request('0x0000000000000000000000000000000000000000', 'latest', callback),
+);
+
+// in 4.x
+batch.add(request1);
```
Comment (nazarhussain) [inline docs/docs/guides/web3_migration_guide.md:48]: Please create a full snippet for 4.x as well. In this script it's not clear from where `request1` came in.

File: docs/docs/guides/web3_migration_guide.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+const request2Promise = batch.add(request2);
+request2Promise.then(response => {
+	console.log(response);
+});
+```
+
+#### execute()
+
+in 4.x execute returns a promise, where 1.x was based on the callbacks of requests passed to `add`
```
Comment (nazarhussain) [inline docs/docs/guides/web3_migration_guide.md:57]: Can we document this as a whole, all of the callbacks are removed except the event listners.

**Review by nazarhussain — CHANGES_REQUESTED** [review]

File: docs/docs/guides/web3_migration_guide.md (hunk 5 above)
Comment (nikoulai) [reply docs/docs/guides/web3_migration_guide.md:57]: Added a comment in `Breaking changes` also.

**Review by nazarhussain — APPROVED** [review]

**Review by jdevcs — APPROVED** [review]

**Review by avkos — APPROVED** [review]

**Review by luu-alex — APPROVED** [review]

**Pull request merged**